### DISCUSSION TITLE
[PSD8] Exercise 2.38

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#163] [Esercizi (A-L)]
**Exercise 2.38** An $M$-bit _thermometer code_ for the number $k$ consists of $k$ 1’s in the least significant bit positions and $M – k$ 0’s in all the more significant bit positions. A _binary-to-thermometer code converter_ has N inputs and $2^N–1$ outputs. It produces a $2^N–1$ bit thermometer code for the number specified by the input. For example, if the input is 110, the output should be 0111111. Design a 3:7 binary-to-thermometer code converter. Give a simplified Boolean equation for each output, and sketch a schematic.

**Comment by Elia-Belli** [comment]
![M bit Thermometer](https://user-images.githubusercontent.com/107944476/211600839-8bbb0c64-2e9a-4814-83fd-f01506fdf3e8.jpg)

**Comment by fraxb04** [comment]
corretto

**Comment by Kikicima** [comment]
Pardon, ho difficoltà a capire come si inizia a risolvere un problema di questo tipo. La tabella della verità immagino sia risolta in base alle equazioni in alto a destra, ma quelle come si ricavano leggendo solo il testo dell'esercizio?

**Comment by fraxb04** [comment]
@Kikicima 
- disegni un componente 3:7 graficamente collegando 3 input e 7 output (giusto per trascrivere il problema in forma grafica).

- ti fai la tabella della verità inserendo agli input i valori 000, 001, 010, 011.... (sono sempre gli stessi per esercizi con tre input), e seguendo ciò che enuncia l'esercizio, trascrivi negli output la sequenza 0111111 agli input 110, quindi logicamente gli output sopra e sotto saranno dei valori che avranno alla cifra più significativa lo 0 ed il resto saranno tutti 1 (come anche scritto nel testo).

- ti ricavi le equazioni booleane degli output:

- o scrivendotele per lungo, per poi semplificare con i teoremi (es: prendi in considerazione y2, guardi (in VERTICALE) quando fa 1, ovvero per gli input A = 1, B = 1, C = 0 e A = 1, B = 1, C = 1, e li trascrivi, quindi: AB!C + ABC = AB(!C + C)= AB(1) = AB),

- o utilizzando le mappe K per averle direttamente semplificate (ti accorgerai che se parti dal calcolarti y1, dovrai solamente aggiungere 1 in un quadretto, senza ricancellare la mappa o farne una per ciascun equazione).

- una volta fatto ciò provi a creare un circuito mettendo prima tutti i vari OR e AND utilizzabili e poi eventualmente andando a semplificarli, poiché l'obiettivo è sempre avere meno porte logiche possibili.

**Comment by Kikicima** [comment]
Grazie mille!
